test/wovn_index_sample_workspace/WOVN.php/src/wovn_helper.php

```diff
@@ -3,11 +3,47 @@
 require_once(__DIR__ . '/wovnio/wovnphp/SSI.php');
 use Wovnio\Wovnphp\SSI;
 
+function reduce_slashes($path) {
+  # Reduces a sequence of slashes to a single slash
+  # e.g. '///./////.///' -> '/././'
+  $i = 0;
+  while($i + 1 < strlen($path)) {
+    if($path[$i] == '/' && $path[$i + 1] == '/') {
+      $path = substr($path, 0, $i) . substr($path, $i + 1, strlen($path));
+    }
+    else {
+      $i++;
+    }
+  }
+  return $path;
+}
+
+function remove_dots_from_path($path) {
+  # Removes '/./ in paths and resolves '/../' in paths
+  # From https://tomnomnom.com/posts/realish-paths-without-realpath
+  # See also http://php.net/manual/en/function.realpath.php#84012
+  $path = reduce_slashes($path);
+
+  $path_parts = explode('/', $path);
+  $tmp_out = array();
+  foreach($path_parts as $part){
+    if ($part == '.') {
+      continue;
+    }
+    if ($part == '..') {
+      array_pop($tmp_out);
+      continue;
+    }
+    $tmp_out[] = $part;
+  }
+  return implode('/', $tmp_out);
+}
+
 function wovn_helper_detect_paths($base_dir, $path_of_url, $files) {
   $request_path = $base_dir . $path_of_url;
-  $local_path = realpath($request_path);
-  $inside_base_dir = $local_path && strpos($local_path, $base_dir) === 0;
-  $local_path = $inside_base_dir ? $local_path : $base_dir;
+  $local_path = remove_dots_from_path($request_path);
+  $local_path = realpath($local_path);
```

Comment:
Now I think it's enough prevent directory traversal.
But I'm not sure 100%, so could you check if file of `$local_path` in directory of `$base_dir` after do `realpath`?
If not, we have to don't accept by security reason.